paasd/src/service.rs

```diff
@@ -27,25 +60,64 @@ impl server_types::ProcessService for ProcessService {
     type GetLogsStream =
         Pin<Box<dyn Stream<Item = Result<LogsResponse, tonic::Status>> + Send + Sync + 'static>>;
 
-    async fn exec(&self, _req: Request<ExecRequest>) -> Result<Response<ExecResponse>, Status> {
-        Err(Status::unimplemented(""))
+    async fn exec(&self, req: Request<ExecRequest>) -> Result<Response<ExecResponse>, Status> {
+        let uid = Self::authenticate(&req)?;
+        let req = req.into_inner();
+        if req.args.is_empty() {
+            return Err(Status::invalid_argument("'args' must be a non-empty array"));
+        }
+        match Process::spawn(&*req.args[0], req.args[1..].iter().map(AsRef::as_ref)) {
+            Ok(p) => {
+                let pid = self.store.insert(&uid, p);
+                Ok(Response::new(ExecResponse {
+                    id: Some(pid.into()),
+                }))
+            }
+            Err(e) => Err(Status::unknown(format!("{}", e))),
+        }
     }
 
     async fn get_logs(
```

Comment:
Doe this always tail? Or can you just fetch current logs?

Reply:
Always tails, as per design. Perhaps not so clearly stated in the design doc.
https://github.com/justinas/paas/blob/5dccc74379aa2ec7d3ab85e833b288d86b0af652/DESIGN.md#getlogs

Reply:
Thanks for the design doc clarification.